packages/flutter_tools/lib/src/android/gradle.dart

```diff
@@ -356,7 +363,30 @@ class AndroidGradleBuilder implements AndroidBuilder {
       command.add('-Ptarget=$target');
     }
     command.addAll(androidBuildInfo.buildInfo.toGradleConfig());
-
+    if (project.manifest.deferredComponents != null) {
```

Comment:
Should there be some more validation here? Could you have deferred components and no component names?

Reply:
That particular case is a valid use case. Developers can include no component names to generate the loading units without packing them into dynamic features during development.